x-pack/plugins/elastic_assistant/server/lib/langchain/execute_custom_llm_chain/index.ts

```diff
@@ -126,7 +127,10 @@ export const callAgentExecutor: AgentExecutor<true | false> = async ({
         returnIntermediateSteps: false,
         agentArgs: {
           // this is important to help LangChain correctly format tool input
-          humanMessageTemplate: `Question: {input}\n\n{agent_scratchpad}`,
+          humanMessageTemplate: `Remember, when you have enough information, always prefix your final JSON output with "Final Answer:"\n\nQuestion: {input}\n\n{agent_scratchpad}.`,
+          memoryPrompts: [new MessagesPlaceholder('chat_history')],
```

Comment:
The `memoryPrompts` change ensures the model receives the chat history. The `humanMessageTemplate` and `suffix` changes are to help with Bedrock parsing